bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractComponentImpl.java

```diff
@@ -137,7 +138,8 @@ private void init() {
     @Override
     public String getId() {
         if (id == null) {
-            this.id = ComponentUtils.getId(this.resource, this.currentPage, null, this.componentContext);
+            String resourceCallerPath = (String) request.getAttribute(FormConstants.RESOURCE_CALLER_PATH);
```

Comment:
you are always typecasting this, what if attribute was not set, then it will throw an exception

Reply:
If an attribute is not set then null value is returned i.e `(String) null` equivalent to `null`